src/Broker.php

```diff
@@ -60,6 +60,25 @@ public function when(callable $condition, $middleware): Broker
         return $this;
     }
 
+    public function paths(array $paths): Broker
+    {
+        $this->middleware[] = [
+            $this->alwaysTrue(),
+            new PathDispatcher($paths, $this)
+        ];
+
+        return $this;
+    }
+
+    public function resolve($middleware): Middleware
```

Comment:
I don't really like this being a public method. What are we gaining by having an interface for resolution, as opposed to a shared trait?

Reply:
Not sure. I was thinking Interface Segregation Principle and "freeing" the `RequestHandler` (and now the `PathDispatcher` as well) from having to know about containers...

Reply:
I don't think that actually helps here. The `ContainerInterface` is a more widely used standard and offers ISP without having to define a local interface that (effectively) does the exact same thing.

Reply:
Fine. So I'll just duplicate the three lines of `resolve()` both in the `RequestHandler` (where they originally come from) and the `PathDispatcher` class?